src/react-queries/index.ts

```diff
@@ -18,6 +31,102 @@ import { thorchainLp } from 'pages/ThorChainLP/queries/queries'
 export const thorchainBlockTimeSeconds = '6.1'
 const thorchainBlockTimeMs = bn(thorchainBlockTimeSeconds).times(1000).toNumber()
 
+const common = createQueryKeys('common', {
+  allowanceCryptoBaseUnit: (
+    assetId: AssetId | undefined,
+    spender: string | undefined,
+    from: string | undefined,
+  ) => ({
+    queryKey: ['allowanceCryptoBaseUnit', assetId, spender, from],
+    queryFn: async () => {
+      if (!assetId) throw new Error('assetId is required')
+      if (!spender) throw new Error('spender is required')
+      if (!from) throw new Error('from address is required')
+
+      if (!isToken(fromAssetId(assetId).assetReference)) return null
+      const supportedEvmChainIds = getSupportedEvmChainIds()
+      if (!supportedEvmChainIds.includes(fromAssetId(assetId).chainId as KnownChainIds)) return null
+
+      const allowanceOnChainCryptoBaseUnit = await getErc20Allowance({
+        address: fromAssetId(assetId).assetReference,
+        spender,
+        from,
+        chainId: fromAssetId(assetId).chainId,
+      })
+
+      return allowanceOnChainCryptoBaseUnit
+    },
+    enabled:
+      assetId &&
+      spender &&
+      from &&
+      getSupportedEvmChainIds().includes(fromAssetId(assetId).chainId as KnownChainIds),
```

Comment:
prefer we call `fromAssetId(assetId)` once and destructure if possible

Reply:
4bd2196d87b72c11466c23fe1fa361fa78a90b14